File: 0061-rotate-list/0061-rotate-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* rotateRight(ListNode* head, int k) {
        int count=0;
        ListNode*temp=head;
        if(!head)return nullptr;
        while(temp){
            count++;
            temp=temp->next;
        }

        k%=count;
        int d=count-k;
        count=0;
        temp=head;
        while(count<d-1){
            count++;
            temp=temp->next;
        }
        ListNode*dummy=new ListNode(-1);
        ListNode*curr=dummy;
        curr->next=temp->next;
        temp->next=nullptr;
        while(curr->next){
            curr=curr->next;
        }
        curr->next=head;
        return dummy->next;
    }
};

File: 0061-rotate-list/0061-rotate-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode rotateRight(ListNode head, int k) {
        if(head==null)return null;
        int count=0;
        ListNode temp=head;
        while(temp!=null){
            count++;
            temp=temp.next;
        }
        if(count==0 || count==k)return head;
        k%=count;
        temp=head;
        for(int i=0;i<count-k-1;i++){
            temp=temp.next;
        }
        ListNode dummy=new ListNode(-1);
        dummy.next=temp.next;
        temp.next=null;
        ListNode curr=dummy;
        while(curr.next!=null){
            curr=curr.next;
        }
        curr.next=head;
        return dummy.next;
    }
}